src/systems/detachable_joint/DetachableJoint.cc

```diff
@@ -234,8 +234,33 @@ void DetachableJoint::PreUpdate(
     }
     else
     {
-      modelEntity = _ecm.EntityByComponents(
-          components::Model(), components::Name(this->childModelName));
+      auto candidateEntities = entitiesFromScopedName(
+        this->childModelName, _ecm);
+
```

Comment:
nit: Please consider filtering this list down by entities which have the Model component. This way, if the entity is found but is not a model, we can indicate that to the user. Otherwise they will end up with a slightly more confusing "Child link ... could not be found" message on line 289.

Reply:
[e8e1b16](https://github.com/gazebosim/gz-sim/pull/1097/commits/e8e1b16976b2774f3f9cbade1eb8c0978aa3279a)